rfcs/text/0408-versioning-types.md

```diff
@@ -0,0 +1,344 @@
+}
+```
+
+```json
+{
+  "https://blockprotocol.org/@alice/types/property-type/name": "The Time Machine",
+  "https://blockprotocol.org/@alice/types/property-type/published-on": "1895-05",
+  "https://blockprotocol.org/@alice/types/property-type/blurb": ...
+}
+```
+
+## Determining Type Compatibility
+
+A key part of the reasoning for picking this approach depends on the ability to determine compatibility between various types.
+
+For the purposes of this section, the following assumptions are applied:
+
+- Any given Schema X is _compatible with_ another schema Y if all possible values that satisfy Schema X also satisfy schema Y (note the directionality)
+- Any given Schema X is _equivalent to_ another Schema Y iff X is _compatible with_ Y _and_ Y is _compatible with_ X
+- All Type schemas have `additionalProperties: false` unless otherwise stated
+- 'Constraints' refer to all JSON schema keywords which affect the validation of data in the Block Protocol, these include (but are not limited to) `type`, `properties`, `minItems`, `maxItems`, `minimum`, `maximum`
+- 'Semantic Annotations' refer to all JSON schema keywords which do not affect the validation of data in the Block Protocol, these include (but are not limited to) `title`, `description`, `examples`, `default`
+- The ordering of constraints does not affect the compatibility of schemas
+
+### Data Types
+
+At the present Data Types represent completely disjoint value spaces, and the base primitive types are all **incompatible** with one another. As the [Non-Primitive Data Types RFC](https://github.com/blockprotocol/blockprotocol/pull/355) continues to be specified, it should comment on how checking compatibility will be affected.
+
+### Property Types
+
+1.  Due to how Property Types are used, they implicitly affect the structure of the data in that they require the existence of a key that matches their URI.
+1.  Property Types define their constraints through a `oneOf` field. This `oneOf` is a collection of a number of values (greater than 1) where each value is a variant of the "Property Values" (defined below).
+
+A Property Type `A` is therefore compatible with another Property Type `B` if and only if
+
+1.  They have the same base URI (which implies they are the same property type but perhaps different versions) and,
+1.  Every element of `A`'s `oneOf` is compatible with one element of `B`'s `oneOf`
+
+The "Property Values" is a recursive definition which refers to the sub-schemas defined within the Property Types meta-schema. It defines the element as being one of the following:
+
+#### Data Type Reference
+
+A Data Type reference is an inline reference to a data-type. A Data Type reference will be compatible with another Data Type reference if the Data Types that are referred to are also compatible.
+
+> _Without_ non-primitive Data Types this implies that the references are to the same Data Type.
+
+#### Property Type Object
+
+A Property Type Object is a JSON object where
```

Comment:
Narrowing it for the avoidance of doubt – many JSON objects which we don't intend to refer to meet the existing definition
```suggestion
A Property Type Object is JSON object defining a schema which is used as `oneOf` the permitted values for a Property Type, where the schema is of `type: "object`, and within its own `properties` definition:
```

Reply:
Much better, thank you